### ISSUE TITLE
perf buffers not closed/flushed after threads end

### ISSUE THREAD

**Issue by rschoene** [tud-zih-energy/lo2s#137]
I start `lo2s` with
```
./lo2s -e task-clock -q -g --standard-metrics -m 1 bash
```
With every incarnation of `make clean` in my NPB-OMP folder within the monitored shell, the number of open file handles for `lo2s` increases by 63.

Without the standard metrics, i.e., 
```
./lo2s -e task-clock -q -g -m 1 bash
```
,the number of file handles increases by 21 per `make clean`

This leads to
```
[54183772675114][pid: 6771][tid: 140057996973888][ERROR]: Failure while adding new process forked from 6995: Too many open files
^[[A
[54184362171933][pid: 6771][tid: 140057996973888][FATAL]: Aborting: Too many open files
```
after short amount of time. It would be great if there was an option to read out buffers and write the OTF2 locations after threads/processes end. This should be enabled by default so that the lo2s does not throw exceptions so fast.

**Comment by rschoene** [comment]
If I increase the soft limit for no of open files, I still have the problem that the amount of mappable memory will run out, even with the `-m 1` flag :disappointed:

**Comment by tilsche** [comment]
This turned out to be the stop-fd pipe of the `PollMonitor` as well as the perf event fd. In issue-137-early-cleanup, we just delete the `ThreadMonitor` when a thread is closed. However, this also causes a `merge_calling_contexts` which may be cause perturbation.

issue-137-early-cleanup-v2 might do the trick...

**Comment by tilsche** [comment]
I consider this fixed by ffc4740c642431d56ff1b6e0d989101f1f568542
If we get other problems with OTF2 files, we will investigate then...
